app/models/user.py

```diff
@@ -3,17 +3,26 @@
 User module for the user table
 """
 from sqlalchemy import Column, Integer, String
-from models import Base
+from models.base_model import Base, Basemodel
+import scrypt
 
-
-class User(Base):
+class User(Base, Basemodel):
     """
     The class representative of the user table
     """
     __tablename__ = 'users'
 
     id = Column(Integer, index=True, primary_key=True)
-    firstname = Column(String(50), index=True)
-    lastname = Column(String(50), index=True)
-    email = Column(String(128), index=True)
+    firstname = Column(String(50), index=True, nullable=False)
+    lastname = Column(String(50), index=True, nullable=False)
+    email = Column(String(128), index=True, nullable=False)
     phone_number = Column(String(10), index=True)
```

Comment:
We might as well as force them to add a phone number as email isn't really a priority here.

We want to target their phone numbers which is quite unique. I'll add it up